integration_test/warehouse/warehouse_test.go

```diff
@@ -1566,6 +1566,63 @@ func TestUploads(t *testing.T) {
 	})
 }
 
+func TestCleanupObjectStorageFiles(t *testing.T) {
+	t.Run("object storage files cleanup", func(t *testing.T) {
+		testcases := []struct {
+			name                      string
+			cleanupObjectStorageFiles bool
+		}{
+			{
+				name:                      "should delete files",
+				cleanupObjectStorageFiles: true,
+			},
+			{
+				name:                      "should not delete files",
+				cleanupObjectStorageFiles: false,
+			},
+		}
+		for _, tc := range testcases {
+			db, minioResource, whClient := setupServer(t, false, func(m map[string]backendconfig.ConfigT, _ *minio.Resource) {
+				m[workspaceID].Sources[0].Destinations[0].Config["cleanupObjectStorageFiles"] = tc.cleanupObjectStorageFiles
+			}, nil)
+			ctx := context.Background()
+			events := 100
+			eventsPayload := strings.Join(lo.RepeatBy(events, func(int) string {
+				return fmt.Sprintf(`{"data":{"id":%q,"user_id":%q,"received_at":"2023-05-12T04:36:50.199Z"},"metadata":{"columns":{"id":"string","user_id":"string","received_at":"datetime"}, "table": "tracks"}}`,
+					uuid.New().String(),
+					uuid.New().String(),
+				)
+			}), "\n")
+			require.NoError(t, whClient.Process(ctx, whclient.StagingFile{
+				WorkspaceID:           workspaceID,
+				SourceID:              sourceID,
+				DestinationID:         destinationID,
+				Location:              prepareStagingFile(t, ctx, minioResource, eventsPayload).ObjectName,
+				TotalEvents:           events,
+				FirstEventAt:          time.Now().Format(misc.RFC3339Milli),
+				LastEventAt:           time.Now().Add(time.Minute * 30).Format(misc.RFC3339Milli),
+				UseRudderStorage:      false,
+				DestinationRevisionID: destinationID,
+				Schema: map[string]map[string]any{
+					"tracks": {
+						"id":          "string",
+						"user_id":     "string",
+						"received_at": "datetime",
+					},
+				},
+			}))
+			requireDownstreamEventsCount(t, ctx, db, fmt.Sprintf("%s.%s", namespace, "tracks"), events)
+			files, err := minioResource.Contents(ctx, "")
+			require.NoError(t, err)
+			if tc.cleanupObjectStorageFiles {
```

Comment:
We can probably pass the `expectedFileCount` as well instead of this if check from `testcases`?